uniffi_bindgen/src/bindings/python/templates/MethodDocsTemplate.py

```diff
@@ -0,0 +1,22 @@
+{% match func.documentation() -%}
+  {% when Some with (docs) %}
+        """
+{% for line in docs.description.lines() %}        {{ line }} 
+{% endfor %}
+
+    {%- if docs.arguments_descriptions.len() > 0 %}
+    
+        Parameters:
+
+{% for arg in func.arguments() %}        - `{{ arg.name() }}`: {{ docs.arguments_descriptions[arg.name()] }}
+{% endfor %} 
+    {%- endif -%}
+
+    {%- match docs.return_description -%}
+      {% when Some with (desc) %}
+
+        Returns: {{ desc }}
+      {%- when None %}
+    {%- endmatch %}        """
+  {% when None %}
+{%- endmatch %}
```

Comment:
I assume this code is good, but the amount of hoops askama is putting you through seems very sad.  One of the experiments I tried in PR #1353 was to implement a new template system that understood whitespace better.  Maybe it would be worth it to extract that code and see if it improved things.

This can be follow up work, but in the meantime I really want some way to verify the results of this template code, because I can't follow it in my head.  That one reason why I like the testing suggestion.